sensor/common/networkflow/manager/manager_impl.go

```diff
@@ -506,27 +511,75 @@ func (m *networkFlowManager) enrichAndSendProcesses() {
 	}
 }
 
+func (m *networkFlowManager) handleContainerNotFound(conn *connection, status *connStatus, enrichedConnections map[networkConnIndicator]timestamp.MicroTS) error {
+	var failReasons error
+	failReasons = multierror.Append(failReasons, fmt.Errorf("ContainerID %s unknown", conn.containerID))
+	timeElapsedSinceFirstSeen := timestamp.Now().ElapsedSince(status.firstSeen)
+	// Expire the connection if the container cannot be found within the clusterEntityResolutionWaitPeriod
+	if timeElapsedSinceFirstSeen > maxContainerResolutionWaitPeriod {
+		if activeConn, found := m.activeConnections[*conn]; found {
+			enrichedConnections[*activeConn] = timestamp.Now()
+			delete(m.activeConnections, *conn)
+			flowMetrics.SetActiveFlowsTotalGauge(len(m.activeConnections))
+			return nil // connection has been enriched
+		}
+		status.rotten = true
+		// Only increment metric once the connection is marked rotten
+		flowMetrics.ContainerIDMisses.Inc()
+		log.Debugf("Unable to fetch deployment information for container %s: no deployment found", conn.containerID)
+	} else {
+		failReasons = multierror.Append(failReasons, fmt.Errorf("time for container resolution (%s) not elapsed yet", maxContainerResolutionWaitPeriod))
+	}
+	return failReasons
```

Comment:
nitpick: just one small suggestion. This is more of a personal opinion - no hard feelings. 😊 

When there is an if/else or nested if->if - it's good to see if that can be flattened. Usually, code turns out to be easier to read and reduces cognitive load (at least, that's what I think).

Some principles that I'm trying to apply:
- invert the `if` statement and move the body outside -> else part or rest of the function goes inside
- use early exits, usually checking for errors first works better (I pick the order of checking by most likely to happen first or least computationally expensive first)

Applied to this function, it would look like this (not tested for correctness):
```suggestion
	timeElapsedSinceFirstSeen := timestamp.Now().ElapsedSince(status.firstSeen)
	if timeElapsedSinceFirstSeen <= maxContainerResolutionWaitPeriod {
		return multierror.Append(fmt.Errorf("ContainerID %s unknown", conn.containerID), fmt.Errorf("time for container resolution (%s) not elapsed yet", maxContainerResolutionWaitPeriod))
	}

	activeConn, found := m.activeConnections[*conn]
	if !found {
		// Expire the connection if the container cannot be found within the clusterEntityResolutionWaitPeriod
		status.rotten = true
		// Only increment metric once the connection is marked rotten
		flowMetrics.ContainerIDMisses.Inc()
		log.Debugf("Unable to fetch deployment information for container %s: no deployment found", conn.containerID)

		return multierror.Append(nil, fmt.Errorf("ContainerID %s unknown", conn.containerID))
	}

	enrichedConnections[*activeConn] = timestamp.Now()
	delete(m.activeConnections, *conn)
	flowMetrics.SetActiveFlowsTotalGauge(len(m.activeConnections))

	// connection has been enriched
	return nil
```

Reply:
Thanks Mladen! I appreciate comments on further potential improvements.
I am aware that the entire enrichment function could use some refactor to simplify and clarify (e.g., is is difficult now to tell if enrichment was successful or failure). In this PR, I wanted to do as little changes as possible, as I wanted to switch epics and make the reviewing faster.

How about me opening a follow-up PR with your suggestion? Would that be okay?

Reply:
Resolving the comment to unblock merging, but the follow up will be there right away.